lib/rucio/tests/test_replica.py

```diff
@@ -851,6 +853,34 @@ def test_client_access_denied_on_delete_replicas(rse_factory, mock_scope, replic
         assert len(replicas) == 1
 
 
+def test_client_list_replicas_on_did_without_replicas(rse_factory, did_factory, replica_client, did_client, root_account):
+    """ REPLICA (CLIENT): dids of type FILE, but without replicas, must be listed with empty pfns and rses"""
+    rse, _ = rse_factory.make_posix_rse()
+    file = did_factory.random_did()
+    dataset = did_factory.make_dataset()
+    container = did_factory.make_container()
+
+    @transactional_session
+    def __add_file_did_without_replica(session=None):
+        models.DataIdentifier(scope=file['scope'], name=file['name'], did_type=DIDType.FILE, bytes=1, adler32='0cc737eb', account=root_account).save(session=session, flush=False)
+
+    __add_file_did_without_replica()
+
+    # make all scopes external
+    file, dataset, container = ({'scope': did['scope'].external, 'name': did['name']} for did in (file, dataset, container))
+
+    did_client.add_files_to_dataset(files=[file], **dataset)
+    did_client.add_datasets_to_container(dsns=[dataset], **container)
+
+    replicas = list(replica_client.list_replicas(dids=[file]))
+    assert len(replicas) == 1
+    assert not replicas[0]['rses']
+    assert not replicas[0]['pfns']
+    # TODO: fix listing dids without replicas from datasets and containers and uncomment the following 2 asserts
```

Comment:
Hmm. Didn't know there is a ticket for that already. I added the test because the behavior of not showing empty replicas was not documented or tested anywhere; but @bari12 told me that it must behave like that.